### ISSUE TITLE
Module name in SourcePos

### ISSUE THREAD

**Issue by nikomi** [osa1/language-lua#16]
When using the annotated parser's `parseText` function the module name in the `SourcePos` is always `<string>`. I would suggest an additional parser function that allows the caller to supply the module name, something like:
```
parseNamedText :: Parser a -> String -> String -> Either ParseError a
parseNamedText p n s = parse p n (llex s)
```
What do you think?

**Comment by osa1** [comment]
The function you're looking for already exists in Parsec:

```
> :m + Text.Parsec
> :t parse
parse
  :: Stream s Data.Functor.Identity.Identity t =>
     Parsec s () a -> SourceName -> s -> Either ParseError a
> :info SourceName
type SourceName = String        -- Defined in ‘Text.Parsec.Pos’
```

Note that `language-lua` exposes Parsec parsers, so you can combine them to have this, for example:

```
> :t parse exp "source name" . llex
parse exp "source name" . llex :: String -> Either ParseError Exp
> parse chunk "source name" . llex $ "invalid syntax ++"
Left "source name" (line 1, column 9):
unexpected (identifier: "syntax",AlexPn 8 1 9)
expecting `.`, `[`, `:`, `{`, string, `(`, `,` or `=`
syntax error
```

**Comment by nikomi** [comment]
Thanx for the hint - that's what I currently do :-)

To do this I have to `import Language.Lua.Annotated.Lexer (llex)` - which I find a bit messy, because it's an internal of `language-lua` I would not normally need to touch. It also requires everyone using the parser and missing source module information to dig for this solution, re-inventing the wheel - adding the enhanced function would make people instantly aware of the possibilities :-)

**Comment by osa1** [comment]
Could you send a pull request with the changes you want?

Please also add the corresponding function to annotated version. E.g. if you add something to Language.Lua.Parser, you should add the corresponding function to Language.Lua.Annotated.Parser.

**Comment by osa1** [comment]
Closed with #18.

**Comment by osa1** [comment]
FYI: New version is available on Hackage.
